tests/bluetooth/tester/src/btp_bas.c

```diff
@@ -0,0 +1,266 @@
+/*
+ * Copyright (c) 2024 Demant A/S
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stddef.h>
+#include <zephyr/bluetooth/gatt.h>
+#include <zephyr/bluetooth/services/bas.h>
+#include "btp/btp.h"
+#include "btp/btp_bas.h"
+
+static uint8_t bas_supported_commands(const void *cmd, uint16_t cmd_len, void *rsp,
+				      uint16_t *rsp_len)
+{
+	struct btp_bas_read_supported_commands_rp *rp = rsp;
+
+	/* octet 0 */
+	tester_set_bit(rp->data, BTP_BAS_READ_SUPPORTED_COMMANDS);
+	tester_set_bit(rp->data, BTP_BAS_SET_BATTERY_LEVEL);
+	tester_set_bit(rp->data, BTP_BAS_BLS_SET_BATTERY_PRESENT);
+	tester_set_bit(rp->data, BTP_BAS_BLS_SET_WIRED_POWER_SOURCE);
+	tester_set_bit(rp->data, BTP_BAS_BLS_SET_WIRELESS_POWER_SOURCE);
+	tester_set_bit(rp->data, BTP_BAS_BLS_SET_BATTERY_CHARGE_STATE);
+	tester_set_bit(rp->data, BTP_BAS_BLS_SET_BATTERY_CHARGE_LEVEL);
+	tester_set_bit(rp->data, BTP_BAS_BLS_SET_BATTERY_CHARGE_TYPE);
+	tester_set_bit(rp->data, BTP_BAS_BLS_SET_CHARGING_FAULT_REASON);
+	tester_set_bit(rp->data, BTP_BAS_BLS_SET_IDENTIFIER);
+	tester_set_bit(rp->data, BTP_BAS_BLS_SET_SERVICE_REQUIRED);
+	tester_set_bit(rp->data, BTP_BAS_BLS_SET_BATTERY_FAULT);
+
+	*rsp_len = sizeof(*rp) + 1;
+
+	return BTP_STATUS_SUCCESS;
+}
+
+static uint8_t btp_bas_set_battery_level(const void *cmd, uint16_t cmd_len, void *rsp,
+					 uint16_t *rsp_len)
+{
+	const struct btp_bas_set_battery_level_cmd *cp =
+		(const struct btp_bas_set_battery_level_cmd *)cmd;
+	uint8_t level = cp->level;
+	int err = bt_bas_set_battery_level(level);
+
+	return err ? BTP_STATUS_FAILED : BTP_STATUS_SUCCESS;
+}
+
+#if defined(CONFIG_BT_BAS_BLS)
+static uint8_t btp_bas_bls_set_battery_present(const void *cmd, uint16_t cmd_len, void *rsp,
+					       uint16_t *rsp_len)
+{
+	const struct btp_bas_bls_set_battery_present_cmd *cp =
+		(const struct btp_bas_bls_set_battery_present_cmd *)cmd;
+	enum bt_bas_bls_battery_present present = cp->present;
```

Comment:
should be validated before pass to function (especially if function is void and doesn't return error)

(same for other handlers)